src/metabase/lib/js.cljs

```diff
@@ -117,28 +119,81 @@
   [a-query stage-number]
   (to-array (lib.order-by/orderable-columns a-query stage-number)))
 
-(defn- display-info-js
-  "Converts the [[metabase.lib.metadata.calculation/display-info]] maps into a JS-friendly form - `camelCase` keys,
-  namespaced keyword values as `\"foo/bar\"` strings, etc.
-  Recurses into nested sequences and maps."
+;; Display-info =====================================================================================================
+;; This is a complicated stack of caches and inner functions, so some guidance is in order.
+;;
+;; The outer surface is `lib.js/display-info` in this file. It has a [[lib.cache/side-channel-cache]], so if
+;; `display-info` is called multiple times on the same opaque CLJS value, it will be cached.
+;;
+;; [[display-info*]] is the inner implementation. It calls [[lib.core/display-info]] to get the CLJS form, then
+;; [[display-info->js]] to convert it to JS.
+;;
+;; JS conversion in the tricky cases (maps and seqs) are handled by separate, LRU-cached functions
+;; [[display-info-map->js]] and [[display-info-seq->js]]. Keywords are converted with [[u/qualified-name]].
+;;
+;; [[display-info-map->js]] converts CLJS maps to JS objects. Keys are converted from `:kebab-case-keywords` to
+;; `"camelCaseStrings"`. Values are recursively converted by [[display-info->js]]. (Note that this passes through the
+;; LRU caches for nested maps and seqs - this is important since many inner pieces are reused across eg. columns.)
+
+;; [[display-info-seq->js]] converts CLJS `sequential?` things to JS arrays, recursively calling [[display-info->js]] on
+;; each element. (This is cached just like map values above.)
+
+;; **Note:** there's an important property here that's worth calling out explicitly. Two columns which are `=` can
+;; be returned by eg. `visible-columns` on two different queries. This is why it's important that the caches used where
+;; `a-query` and `stage-number` are provided ([[display-info]]) are side channel caches that only cache for a specific
+;; instance, not for any `=` value. The different queries might lead to different display names or other details.
+;; The CLJS -> JS conversion doesn't know about queries, so it uses `=`-based LRU caches.
+(declare ^:private display-info->js)
+
+(defn- display-info-map->js* [x]
+  (reduce (fn [obj [cljs-key cljs-val]]
+            (set! js/window.__DIJS_recurses (inc js/window.__DIJS_recurses))
```

Comment:
Is this for debugging only?

Reply:
Yes, I missed some of that. I've removed it now.